shortfin/python/shortfin_apps/sd/simple_client.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import json
+import requests
+import argparse
+import base64
+import time
+import asyncio
+import aiohttp
+import sys
+import os
+
+from datetime import datetime as dt
+from PIL import Image
+
+sample_request = {
+    "prompt": [
+        " a cat under the snow with blue eyes, covered by snow, cinematic style, medium shot, professional photo, animal",
+    ],
+    "neg_prompt": ["Watermark, blurry, oversaturated, low resolution, pollution"],
+    "height": [1024],
+    "width": [1024],
+    "steps": [20],
+    "guidance_scale": [7.5],
+    "seed": [0],
+    "output_type": ["base64"],
+    "rid": ["string"],
+}
+
+
+def bytes_to_img(bytes, idx=0, width=1024, height=1024, outputdir="./gen_imgs"):
+    timestamp = dt.now().strftime("%Y-%m-%d_%H-%M-%S")
+    image = Image.frombytes(
+        mode="RGB", size=(width, height), data=base64.b64decode(bytes)
+    )
+    if not os.path.isdir(outputdir):
+        os.mkdir(outputdir)
+    im_path = os.path.join(outputdir, f"shortfin_sd_output_{timestamp}_{idx}.png")
+    image.save(im_path)
+    print(f"Saved to {im_path}")
+
+
+def get_batched(request, arg, idx):
+    if isinstance(request[arg], list):
+        if len(request[arg]) == 1:
+            indexed = request[arg][0]
+        else:
+            indexed = request[arg][idx]
+    else:
+        indexed = request[arg]
+    return indexed
+
+
+async def send_request(session, rep, args, data):
+    try:
+        print("Sending request batch #", rep)
+        url = f"http://0.0.0.0:{args.port}/generate"
+        start = time.time()
+        async with session.post(url, json=data) as response:
+            end = time.time()
+            # Check if the response was successful
+            if response.status == 200:
+                response.raise_for_status()  # Raise an error for bad responses
+                timestamp = dt.now().strftime("%Y-%m-%d_%H-%M-%S")
+                res_json = await response.json(content_type=None)
+                if args.save:
+                    for idx, item in enumerate(res_json["images"]):
+                        width = get_batched(data, "width", idx)
+                        height = get_batched(data, "height", idx)
+                        print("Saving response as image...")
+                        bytes_to_img(
+                            item.encode("utf-8"), idx, width, height, args.outputdir
+                        )
+                latency = end - start
+                print("Responses processed.")
+                return latency, len(data["prompt"])
+            else:
+                print(f"Error: Received {response.status} from server")
+                raise Exception
+    except Exception as e:
+        print(f"Request failed: {e}")
```

Comment:
This except clause seems redundant with the internal one. Based on docs for async with you already have a try except inside there right?